### PR TITLE
#2139 Upgrade Cobrix to v2.6.0

### LINKED ISSUE #2139: Update to newest verson of Cobrix
## Background
[Cobrix](https://github.com/AbsaOSS/cobrix) is in higher version than the used one.

## Feature
Update to the newest version of [Cobrix](https://github.com/AbsaOSS/cobrix) available. There seem to be some interface changes.

### PR DESCRIPTION
**Pull request by jakipatryk**
- upgraded to v2.6.0 Cobrix
- as Cobrix now adds some additional int-metadata to some fields:
  - enabled int-metadata input for Spark schema to Enceladus schema converter
  - made custom Equality for StructType comparison without metadata field for SchemaParserSuite
- updated the way record format is provided to Cobrix

Closes #2139

### CODE REVIEW

File: spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationPropertiesProvider.scala [hunk 1]
```diff
@@ -130,12 +130,21 @@ class StandardizationPropertiesProvider {
       val isXcomOpt = if (cobolOptions.isXcom) Some(true) else None
       val isTextOpt = if (cobolOptions.isText) Some(true) else None
       val isAscii = cobolOptions.encoding.exists(_.equalsIgnoreCase("ascii"))
+
```
Comment (benedeki) [inline spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationPropertiesProvider.scala:133]: With these changes and removal of `"is_record_sequence"` and `"is_text"` from the `HashMap` I believe `isXcomOpt` and `isTextOpt` doesn't make much sense anymore.

Comment (jakipatryk) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationPropertiesProvider.scala]: Took me some time to understand what you mean (initially I thought you were talking about `isXcom` and `isText` from `CobolOptions`) :D I agree, removed them.

File: rest-api/src/test/scala/za/co/absa/enceladus/rest_api/utils/parsers/SchemaParserSuite.scala [hunk 2]
```diff
@@ -30,6 +31,28 @@ import za.co.absa.enceladus.rest_api.utils.SchemaType
 import za.co.absa.enceladus.rest_api.utils.converters.SparkEnceladusSchemaConvertor
 
 class SchemaParserSuite extends AnyWordSpec with Matchers with MockitoSugar with Inside {
+
+  implicit val structTypeEqualityIgnoreMetadata: Equality[StructType] = new Equality[StructType] {
```
[the remark this answers is not in the document]
Comment (jakipatryk) [reply rest-api/src/test/scala/za/co/absa/enceladus/rest_api/utils/parsers/SchemaParserSuite.scala]: Made private and added a comment.

File: rest-api/src/test/scala/za/co/absa/enceladus/rest_api/utils/parsers/SchemaParserSuite.scala [hunk 3]
```diff
@@ -32,7 +32,11 @@ import za.co.absa.enceladus.rest_api.utils.converters.SparkEnceladusSchemaConver
 
 class SchemaParserSuite extends AnyWordSpec with Matchers with MockitoSugar with Inside {
 
-  implicit val structTypeEqualityIgnoreMetadata: Equality[StructType] = new Equality[StructType] {
+  // `metadata` field is considered to be implementation detail of a specific, external parser;
```
Comment (benedeki) [inline rest-api/src/test/scala/za/co/absa/enceladus/rest_api/utils/parsers/SchemaParserSuite.scala:35]: Love the comment 👍

File: spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationPropertiesProvider.scala (hunk 1 above)
Comment (benedeki) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationPropertiesProvider.scala]: Sorry, if it was not clear. Next time don't hesitate to ask. 😉

**Review by dk1844 — APPROVED** [review]
LGTM (just read the code). I have also retried the Jenkins run (unrelated failures on timeouts)

**Comment by dk1844** [comment]
Merged in lieu of @jakipatryk

**Pull request merged**